ddtrace/tracer/tracer.go

```diff
@@ -133,6 +134,7 @@ const payloadQueueSize = 1000
 func newUnstartedTracer(opts ...StartOption) *tracer {
 	c := new(config)
 	defaults(c)
+	globalconfig.SetServiceName(c.serviceName)
```

Comment:
With all my shame, I take back my previous comment, this looks quite off place here and seemed much better in `defaults`.